dynamiqs/result.py

```diff
@@ -120,3 +120,48 @@ class MEResult(Result):
         gradient _(Gradient)_: Gradient used.
         options _(Options)_: Options used.
     """
+
+
+class SMESaved(Saved):
+    Isave: Array
+    wiener: PyTree
+
+
+class SMEResult(Result):
+    """Result of the SME integration.
+
+    Attributes:
+        states _(array of shape (nH?, nrho0?, ntrajs, ntsave, n, n))_: Saved states.
```

Comment:
It's the same in `MEResult`, but I guess we could also batch over jump operators. In any case, we'll have to properly document this at some point: it's very much related to the discussion on TimeArrays.

I'm thinking a notation like `(nH..., nLs..., nrho0..., ntrajs, ntsave, n, n)` could be more explicit than what we currently have.

Reply:
Yes, it's tracked in https://linear.app/dynamiqs/issue/DYN-73/support-generalized-batching.

> I'm thinking a notation like `(nH..., nLs..., nrho0..., ntrajs, ntsave, n, n)` could be more explicit than what we currently have.
Once we've implented the ND batching yes, I like `H: (...H, n, n)` where I read `...H` as "all the other dimensions of H`.

Note that documenting the batching over `jump_ops` is a bit tricky because technically it is `(...L1, ...L2, ..., ...L_nL)`.